torch/_inductor/codecache.py

```diff
@@ -491,6 +478,16 @@ def _reduce_tensor(t):
         return (_ident, (metadata,))
 
 
+def _reduce_symint(s):
+    """
+    See FxGraphCachePickler. Custom reducer to pickle SymInts.
+    """
+    # For hashing purposes, we only care about the name of the symbol and
```

Comment:
It might be useful in the future to canonicalize symints so that `hash(tensor([s1, s1, s2])) == hash(tensor([s3, s3, s4]))`.. not needed now

Reply:
It's not clear to me that this is sound, given https://github.com/pytorch/pytorch/pull/111421/files#diff-c9b517f8db609ffa866804dfa2689188a4fee20abacaa0b0dca91625c1b5cb8dR705

if you say [s0, s1] == [s4, s3], you have to make sure that you know how to flip the SymInt arguments when eval'ing the guards. That sounds difficult.